core/net/rpl/rpl-timers.c

```diff
@@ -115,7 +115,7 @@ new_dio_interval(rpl_instance_t *instance)
   time = 1UL << instance->dio_intcurrent;
 
   /* Convert from milliseconds to CLOCK_TICKS. */
-  ticks = (time * CLOCK_SECOND) / 1000;
+  ticks = time / ( 1000 / CLOCK_SECOND);
```

Comment:
Some platforms have CLOCK_CONF_SECOND set to 1024, leading to division by zero after this change. Also the change makes the timer a little bit inaccurate for small values of `time`. BTW, it seems that with the default values of `RPL_CONF_DIO_INTERVAL_MIN` and `RPL_CONF_DIO_INTERVAL_DOUBLINGS` the existing code should work fine on all platforms, so the problem should only appear when those are increased?